python/tk_desktop/desktop_engine_site_implementation.py

```diff
@@ -231,6 +249,79 @@ def run(self, splash=None, version=None):
         result = app.exec_()
         return result
 
+    def uses_legacy_authentication(self):
+        """
+        Returns if the Desktop's bootrap code uses the tk-framework-login for
+        logging in.
+
+        :returns: True the bootstrap logic is older than 1.1.0, False otherwise.
+        """
+        return StrictVersion(self.app_version) < StrictVersion("1.1.0")
+
+    def create_legacy_login_instance(self):
+        """
+        Creates the pre-bootstrap-1.1.0 ShotgunLogin used to authenticate.
+
+        :raises ImportError: Thrown if the module is not available.
+
+        :returns: A tk-framework-login's ShotgunLogin instance.
+        """
+        try:
+            from python import ShotgunLogin
+        except ImportError:
+            self.log_exception("Could not import tk-framework-login")
+            raise
+        else:
+            return ShotgunLogin.get_instance_for_namespace("tk-desktop")
+
+    def _assert_authentication_fault_exists(self, splash):
+        """
+        Asserts that we importing the right version of shotgun_api3. If we are
+        not, the user will be shown a countdown and the app will be restarted.
+        """
+        from tank_vendor import shotgun_api3
+        try:
+            # Try to access the new AuthenticationFault exception. If we are
+            # running a pre-1.0.2 version of Desktop, Desktop will not have been
+            # rebooted after the core upgrade and we'll get the pre 0.16.0
+            # shotgun_api3 module, which doesn't have the AuthenticationFault
+            # class.
+            shotgun_api3.AuthenticationFault
```

Comment:
Should we be worried about this check? Are we essentially using this to determine if the a core distribution is v0.16 or earlier? Would it make sense to for example read the `info.yml` file that is at the top level of the core package? This is what we normally use to check the version of the core.